extension/utilities/PluginApi.js

```diff
@@ -0,0 +1,128 @@
+const crypto = require('crypto')
+
+class PluginApi {
+  constructor (context) {
+    this.host = context.config.shopgatePluginUrl
+    this.request = context.tracedRequest('ShopgateProjectShopgatePluginAPI')
+    this.logger = context.log
+    this.customerNumber = context.config.shopgateCustomerNumber
+    this.shopNumber = context.config.shopgateShopNumber
+    this.apiKey = context.config.shopgateApiKey
+    this.hash = crypto.createHash('sha1')
+  }
+
+  haveInformationForRequest () {
+    return this.host && this.shopNumber && this.apiKey && this.customerNumber
+  }
+
+  makeAuthTokens () {
+    const timeStamp = Math.floor(Date.now() / 1000)
+    const authUser = `${this.customerNumber}-${timeStamp}`
+    const tokenData = `SPA-${authUser}-${this.apiKey}`
+    this.hash.update(tokenData)
+    const token = this.hash.digest('hex')
+    return { authUser, token }
+  }
+
+  /**
+   * Sanitizes params for use in logging
+   * @param {*} params Params to be logged
+   * @return {*|{}}
+   */
+  sanitizeForLogging (params) {
+    let sanitizedForLoggingBody = params
+    if (typeof params === 'object') {
```

Comment:
not sure if it is relevant here, but keep in mind:
null is also an object `typeof null === 'object'`

Reply:
added check for the truthyness of the params in addition to its type.